controllers/generic_controller.go

```diff
@@ -379,16 +379,15 @@ func (gr *GenericReconciler) reconcileDelete(ctx context.Context, metaObj azcore
 		return ctrl.Result{}, fmt.Errorf("unable to transform to resource with: %w", err)
 	}
 
-	switch resource.ProvisioningState {
-	case zips.DeletingProvisioningState:
+	if resource.ProvisioningState == zips.DeletingProvisioningState {
 		msg := "deleting... checking for updated state"
 		gr.Recorder.Event(metaObj, v1.EventTypeNormal, "ResourceDeleteInProgress", msg)
 		return gr.updateFromNonTerminalDeleteState(ctx, resource, metaObj)
-	default:
-		msg := fmt.Sprintf("start deleting resource in state %q", resource.ProvisioningState)
-		gr.Recorder.Event(metaObj, v1.EventTypeNormal, "ResourceDeleteStart", msg)
-		return gr.startDeleteOfResource(ctx, resource, metaObj)
 	}
```

Comment:
Interesting that the linter objects to simple **switch** statements.

Reply:
it objects to non-exhaustive switches. in this case it was simpler to convert it to `if-else`